### ISSUE TITLE
Transpiling local package that already has react in node modules

### ISSUE THREAD

**Issue by iwarner** [martpie/next-transpile-modules#50]
- [X] I HAVE READ [THE FAQ](https://github.com/martpie/next-transpile-modules#faq) AND MY PROBLEM WAS NOT DESCRIBED THERE

**Are you trying to transpile a local package or an npm package?**
Local

**Describe the bug**
- PackageB brought in via yarn link to PackageA
- transpileModules: ['PackageB'],
- PackageB already has node_modules including react
- Get error with new hooks Invalid hook call. Hooks can only be called inside of the body of a function component
- This is due to PackageA and B both having react.
- I have tried externals - but to no avail

```
    webpack (config, options) {
      config.externals = config.externals || []

      config.externals = {
        react: {
          root: 'React',
          commonjs2: 'react',
          commonjs: 'react',
          amd: 'react'
        },
        'react-dom': {
          root: 'ReactDOM',
          commonjs2: 'react-dom',
          commonjs: 'react-dom',
          amd: 'react-dom'
        }
      }
```

and 

config.resolve.alias.react = path.resolve('./node_modules/react')

**To Reproduce**
Hard to reproduce as need linked packages
Steps above are quite easy to understand

**Expected behavior**
Externals should work on the transpiled module only, so the transpiled module can use the packageA react

**Setup**

- Next.js version: 9
- `next-transpile-modules` version: 2.3.1
- Node.js version: 12
- `npm` version: 6
- Operating System: oxs

**Additional context**

**Comment by martpie** [comment]
Hello, sorry for the late reply.

Are you saying 

```config.resolve.alias['react'] = path.resolve(__dirname, '.', 'node_modules', 'react');```

is not solving this issue?

**Comment by HosseinAgha** [comment]
@martpie this makes using this package with react hooks impossible. Here is a detailed explanation for this bug:  
If I transpile a module that requires react internally, It will add that instance of react to the server bundle. In other words the `node_modules` directory of the transpiled modules is not added to webpack `externals` on server side.  
This causes multiple instances of react to be loaded in server side render and [famous react hooks error](https://reactjs.org/warnings/invalid-hook-call-warning.html).
I use NextJS `9.1.4`.

**Comment by HosseinAgha** [comment]
I investigated the code. I'm pretty sure everything is OK with your code as it will not remove react of transpiled module from externals. [This line](https://github.com/zeit/next.js/blob/332852bf0125a7c6ef0393fe21135353138653d6/packages/next/build/webpack-config.ts#L474) in the nextjs externals function creates this issue.  
I've aliased react to project's react in webpack (the official nextjs recommendation for these libraries that have side effects) maybe aliasing messes up your plugin's logic.  
For future reference if you get `Hooks can only be called inside the body of a function component.` when using this module. First make sure you have aliased `react` and `react-dom` and every library that needs a single instance using the [next-preact](https://github.com/zeit/next-plugins/tree/master/packages/next-preact)'s approach (substitute preact with `require.resolve('libname')`). Finally don't forget to add those libraries to externals __after__ the NextJS default externals (Take a look at [this line](https://github.com/zeit/next-plugins/blob/master/packages/next-preact/index.js#L10))  
Finally We need open an issue in Next repo or find a way to fix it here.
